encoding/src/serde_byte_array.rs

```diff
@@ -5,48 +5,41 @@ use crate::BYTE_ARRAY_MAX_LEN;
 use serde::{de, ser, Deserializer, Serializer};
 use serde_bytes::{Deserialize, Serialize};
 
-/// serde_bytes with max length check
-pub mod serde_byte_array {
-    use super::*;
+/// checked if input > `crate::BYTE_ARRAY_MAX_LEN`
+pub fn serialize<T, S>(bytes: &T, serializer: S) -> Result<S::Ok, S::Error>
+where
+    T: ?Sized + Serialize + AsRef<[u8]>,
+    S: Serializer,
+{
+    if bytes.as_ref().len() > BYTE_ARRAY_MAX_LEN {
+        return Err(ser::Error::custom::<String>(
```

Comment:
0. I can add a `SerdeError` in crate `encoding`

```
pub enum SerdeError {
    ArrayExceedsMaxLength,
}
```

1. use previous `encoding::Error`

or I can add an `Any` item in [`CodecProtocol`](https://github.com/ChainSafe/forest/blob/e66d0f9e2f810752bd3330d9ae2370f7197b234b/encoding/src/errors.rs#L50) to describe this error

```rust
pub enum CodecProtocol {
  // use any since this error occurs in serialize && deserialize
  //
  // add this line ⬇️
  Any,
  Cbor,
}
```

and the error would be like

```rust
let arr_limit_err = encoding::Error {
  description: "Array exceeds max length".into(),
  protocal: CodecProtocol::Any,
}
```

Reply:
Both are fine to me, perhaps having it is `ArrayExceedsMaxLength(usize)` would make it easy to pass the offending length to `Display`.

Reply:
https://github.com/ChainSafe/forest/pull/1434/commits/9986f52e7438eb4f2bf01f279a22b0ea273f4b55

Reply:
Thank you!